### ISSUE TITLE
Noticeable delay each time in-product release notes editor is made active.

### ISSUE THREAD

**Issue by gregvanl** [Microsoft/vscode#13257]
- VSCode Version:1.6.0 insiders
- OS Version: Windows

Steps to Reproduce:
1. Follow test plan to enable in product release notes in insiders https://github.com/Microsoft/vscode/issues/12520
2. Help > Release Notes
3. Note delay (~1.5 seconds) when page is initially displayed.
4. Switch to a different editor and then go back to release notes.

The release notes editor is re-rendered and there is the same delay each time. 
We should look into caching the rendered page since release notes are unlikely to change during the user's reading of the release notes. We don't need to cache release notes across sessions so the user will pick up any release note changes when they first open them (Help > Release Notes).

**Comment by gregvanl** [comment]
On my home wifi this can take up to 5 seconds (depending on what the kids are streaming).

**Comment by olingern** [comment]
I'm interested in helping out with this. A couple of questions for @joaomoreno @Tyriar 

Idea: 
Place a hook inside of `loadReleaseNotes` in `src/vs/workbench/electron-browser/update.ts` to see if a cached markup file exists, if it does load it instead of retrieving from `https://code.visualstudio.com`. Otherwise, load from url and write markup response to disk.

1) Frequency of caching: Are release notes final, or do they often update without regards to a release?
2) Location of file: I'm newer to the directory structure and thought  `out/vs/workbench/electron-browser/cache/release-notes.md` may be a good place.

**Comment by joaomoreno** [comment]
@olingern You shouldn't write them to disk, but just keep them in memory. It's OK if they take up time loading once Code restarts.

**Comment by olingern** [comment]
@joaomoreno This one is a bit tricky as it doesn't seem I can test through the development build. Do you have any pointers for creating a build where I can see a full list of menu items?

<img width="813" alt="screen shot 2016-10-27 at 8 42 40 pm" src="https://cloud.githubusercontent.com/assets/1470297/19790444/3954392e-9c86-11e6-82a7-20d82429762e.png">

**Comment by joaomoreno** [comment]
Yeah, you have to add to the top level `product.json` the following key:

```json
	"releaseNotesUrl": "https://go.microsoft.com/fwlink/?LinkId=724002"
```

**Comment by olingern** [comment]
@joaomoreno Thanks!

I have a PR open, #15050, with a working fix, but a couple of issues:
- linting
  The gulp hygiene task is telling me my file ( and all files ) aren't formatted. I saw a couple of other issues in the repo chatting about this. I installed **typescript-formatter** and ran `tsfmt -r update.ts`, but it produced an error

<img width="941" alt="screen shot 2016-11-05 at 7 48 09 pm" src="https://cloud.githubusercontent.com/assets/1470297/20034347/f84ebc30-a390-11e6-92f0-aba656abd9e0.png">

Running `gulp hygiene` tells me all files in my project aren't formatted as well:

<img width="910" alt="screen shot 2016-11-05 at 6 18 59 pm" src="https://cloud.githubusercontent.com/assets/1470297/20034349/199d9514-a391-11e6-86fb-fece4509db06.png">

- Tests

It wasn't clear to me where tests for this feature would live. Based upon where other tests live, it looks like `src/vs/workbench/test/electron-browser/update.ts` is where this should live, but the `electron-browser` test directory doesn't exist, at the moment.
